Upgrade Software doc improvements (#1794)
astronomer/docs: software/upgrade-astronomer.md

NAMESPACE=<astronomer-platform-namespace>
ASTRO_VERSION=<astronomer-version>

helm repo add astronomer https://helm.astronomer.io
helm repo update

# upgradeDeployments false ensures that Airflow charts are not upgraded when this script is ran
# If you deployed a config change that is intended to reconfigure something inside Airflow,
# then you may set this value to "true" instead. When it is "true", then each Airflow chart will
# restart. Note that some stable version upgrades require setting this value to true regardless of your own configuration.
# If you are currently on Astronomer Software 0.25, 0.26, or 0.27, you must upgrade to version 0.28 before upgrading to 0.29. A direct upgrade to 0.29 from a version lower than 0.28 is not possible.
helm upgrade --namespace $NAMESPACE \
            -f ./config.yaml \
            --reset-values \
            --version $ASTRO_VERSION \

>             --timeout 20m \

Author: ```suggestion
            -- debug \
            --timeout 20m \
```

To avoid extended service disruptions, Astronomer recommends upgrading Astronomer Software to a compatible version before you upgrade Kubernetes. To view Astronomer Software and Kubernetes compatibility information, see [Version compatibility reference for Astronomer Software](version-compatibility-reference.md#astronomer-software).

### Supported upgrade paths 

> If you're upgrading through multiple Astronomer Software versions in a single upgrade process, review the following table to ensure that you follow the correct upgrade path to your target version. If your combination of **Current version** and **Target version** isn't listed, then it's safe to directly upgrade from your current version to the target version. 

Reviewer: ```suggestion
If your upgrade path includes multiple, successive Astronomer Software versions, review the following table to ensure that you're following the correct upgrade path. If your combination of **Current version** and **Target version** isn't listed, you can upgrade directly from your current version to the target version. 
```
